SharpBucket/Utility/UrlHelper.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Text;
+
+namespace SharpBucket.Utility
+{
+    internal static class UrlHelper
+    {
+        /// <summary>
+        /// Concat url path segments until the first null or empty segment.
+        /// </summary>
+        public static string ConcatPathSegments(params string[] pathSegments)
```

Comment:
I wonder if this should go in an UrlHelper class like I did, or if it's better to declare this as an extension method in StringExtensions

@MitjaBezensek have you a preference on how to expose such method ?